vendor/github.com/docker/libnetwork/endpoint.go

```diff
@@ -528,7 +528,8 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 	}()
 
 	if sb.needDefaultGW() && sb.getEndpointInGWNetwork() == nil {
-		return sb.setupDefaultGW()
+		err = sb.setupDefaultGW()
+		return err
```

Comment:
This is part of another change brought in by the vendoring. 
It is to make sure the existing defer function which performs the cleanup in case of `err != nil` is being executed.